automatic triggering of Ygor Package Updates

VolkerHBZ opened the pull request:
at 6:00 am when
config.gokb.packageUpdate_enabled = true and
config.gokb.ygorUrl is set

VolkerHBZ committed:
automatic triggering of Ygor Package Updates at 6:00 am when
config.gokb.packageUpdate_enabled = true and
config.gokb.ygorUrl is set

hornmo commented:
The jobs should also consider the defined frequency of each source. The new config items should be added with default values (false & null) to the global config in application.yml.

VolkerHBZ committed:
added configuration options to application.yml
checking the update frequency against lastRun before sending the update request to Ygor
the frequency string consists of a number followed by a letter with the letter representing the units (Day, Week, Month, Quarter, Half, Year) like 1W or 5M. whitespaces are ignored, the number and letter are both mandatory.

VolkerHBZ committed:
waiting for the Ygor enrichment process to finish before starting the next package

VolkerHBZ committed:
moved the actual update logic to the PackageService

hornmo committed:
integration source handling, synchronize package updates

hornmo committed:
fix merge issues

hornmo committed:
default cg status filter, fix componentlookup sorting hql

hornmo committed:
fix concurrent

hornmo committed:
source namespace & checks

philboeselager commented on server/gokbg3/grails-app/domain/org/gokb/cred/Source.groovy:
```diff
@@ -142,4 +144,37 @@ class Source extends KBComponent {
+      int days
+      if (matcher.find()) {
+        def length = [
+          D: 1,
```
This is OK for now, but would need to be refined. Chosen the option "M1" for example, the user would expect the update to happen at the very same day of each month.

VolkerHBZ replied:
M1 won't pass the parser, as it expects the number *before* the letter. Maybe the user expects the update at the same day-of-month, though I highly doubt that he or she notices the slight shift of 2 days max, within one month. There's no way of entering that specific day-of-month for the update anyway. The best solution might actually be to use your _inverted_ syntax, with "M1" meaning each first day of the month or "W3" for the thirds day each week.

philboeselager commented on server/gokbg3/src/main/groovy/com/k_int/ClassUtils.groovy:
```diff
@@ -197,4 +197,14 @@ class ClassUtils {
+
+  public static boolean setBooleanIfDifferent(obj, prop, value) {
+    if ((obj != null) && (prop != null) && (value != null)) {
+      if (obj[prop] != value) {
```
The check in 203 is not necessary.

philboeselager reviewed (approved):
The intervals table (first comment) should be refined later.
The unnecessary check (second comment) can be removed at next occasion.
